### PR TITLE
Implementing More Datetime Primitives

### PR DESCRIPTION
**Pull request by sbadithe**
Working on adding more datetime primitives: 

- DaysInMonth (issue#2067)
- Quarter (issue#2066)
- IsLeapYear (issue#2066)
- IsQuarterEnd (issue#2065) 
- IsQuarterStart (issue#2064)

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -6,11 +6,12 @@ Release Notes
...
-        * Implement Quarter primitive (:pr:`2117`)
-        * Implement IsLeapYear primitive (:pr:`2117`)
-        * Implement IsQuarterEnd primitive (:pr:`2117`)
-        * Implement IsQuarterStart primitive (:pr:`2117`)
+        * Implement ``DaysInMonth`` primitive (:pr:`2117`)
+        * Implement ``Quarter`` primitive (:pr:`2117`)
+        * Implement ``IsLeapYear`` primitive (:pr:`2117`)
+        * Implement ``IsQuarterEnd`` primitive (:pr:`2117`)
+        * Implement ``IsQuarterStart`` primitive (:pr:`2117`)
+        * Implement ``DayOfYear`` primitive (:pr:`2110`)
```
Comment (gsheni) [inline docs/source/release_notes.rst:14]:
```suggestion
        * Add ``DaysInMonth``, ``Quarter`` , ``IsLeapYear`` , ``IsQuarterEnd`` , ``IsQuarterStart``, ``DayOfYear`` transform primitives (:pr:`2110`) 
```

**Comment by gsheni** [comment]
Do we need to worry about timezone aware datetimes with any of these primitives?

**Comment by sbadithe** [comment]
> Do we need to worry about timezone aware datetimes with any of these primitives?

I do not believe so

File: featuretools/tests/primitive_tests/test_transform_primitive.py [hunk 2]
```diff
@@ -113,6 +118,46 @@ def test_day_of_year():
     np.testing.assert_array_equal(days_of_year, correct_days)
 
 
+def test_days_in_month():
+    dim = DaysInMonth()
+    dates = pd.Series([datetime(2010, 1, 1)])
```
Comment (dvreed77) [inline featuretools/tests/primitive_tests/test_transform_primitive.py:123]: add more test cases. Add february of leap year and not leap year

**Review by dvreed77 — APPROVED** [review]
lgtm

**Pull request merged**